### ISSUE TITLE
Loss of ec2 instance causes terraform to fail due to eni- in security group configs.

### ISSUE THREAD

**Issue by tburt11** [terraform-providers/terraform-provider-aws#1796]
When an instance created in a terraform plan is manually deleted (or is killed off or disappears) then subsequent runs of the plan will fail if there are resources in that plan that were dependent on the eni that was part of the instance.  Security Groups for example.

```
* aws_network_interface_sg_attachment.common_target[2]: aws_network_interface_sg_attachment.common_target.2: InvalidNetworkInterfaceID.NotFound: The networkInterface ID 'eni-2fdae501' does not exist
```

The only way to rescue the plan is to create a temporary ec2 instance and search and replace the terraform.state file all occurances of the bad eni with the temporary eni.

With a valid (but unrelated) eni in the security groups config, then terraform can perform a destroy and a subsequent apply.

Would it be acceptable to allow a destroy of security groups when the eni is not valid?

### Terraform Version

Terraform v0.10.4

### Affected Resource(s)
all

### Expected Behavior
terraform plan  -> success

### Actual Behavior
terraform pan -> fail

### Steps to Reproduce
Create a plan that creates one instance and attach security groups

Manually delete the instance.

Run terraform plan.  -> Get errors that eni does not exist.

```
* aws_network_interface_sg_attachment.common_target[2]: aws_network_interface_sg_attachment.common_target.2: InvalidNetworkInterfaceID.NotFound: The networkInterface ID 'eni-2fdae501' does not exist
```

Substitute a valid eni from any ec2 instance in the state file and then a destroy may be completed.

**Comment by bflad** [comment]
Pull request submitted: #6048

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.39.0 of the AWS provider, likely later today.
